deploy-0chain/action.yml

```diff
@@ -176,61 +183,141 @@ runs:
 
-        echo "Waiting for 0Chain deployment to finish..."
-        sleep 70
-        counter=0;
-        while [ $counter -lt 30 ] && kubectl -n ${{ env.NAMESPACE }} get pods --kubeconfig ./kube/${{ env.NAMESPACE }}-config | grep blobber-stake | grep -vi Completed; do
-
-          if ( kubectl -n ${{ env.NAMESPACE }} get pods --kubeconfig ./kube/${{ env.NAMESPACE }}-config | grep blobber-stake | grep -i Error ); then
-            echo "::$ANNOTATION_LEVEL title=0Chain network deployment failed on $RETRY_ATTEMPT attempt::blobber stake failed on $RETRY_ATTEMPT attempt"
-            exit 1
-          fi
+    - name: "Stake blobbers using above built binaries"
+      shell: 'script --return --quiet --command "bash {0}"'
+      run: |
+        echo 'block_worker: https://dev.0chain.net/dns
+        signature_scheme: bls0chain
+        min_submit: 50 # in percentage
+        min_confirmation: 50 # in percentage
+        confirmation_chain_length: 3
+        ethereum_node_url: https://ropsten.infura.io/v3/22cb2849f5f74b8599f3dc2a23085bd4
+        max_txn_query: 15
+        query_sleep_time: 3
+        ' > zbox_config.yaml
+        sed "s,block_worker:.*,block_worker: https://${{ env.NETWORK_URL }}/dns," -i ./zbox_config.yaml
+        echo '#!/bin/bash
+        echo
+        echo "==========================================================="
+        echo "STAKING BLOBBERS USING ZWALLETCLI & ZBOXCLI BINARIES"
+        echo "==========================================================="
+        echo
+        exec 6>&1 # Save stdout to file descriptor 6
+        function retry {
+            local n=1
+            local max=3
+            local delay=5
+            while true; do
+            "$@" && break || {
+                if [[ $n -lt $max ]]; then
+                    ((n++))
+                    echo "Command failed. Attempt $n/$max:"
+                    sleep $delay;
+                else
+                    exec 1>&6 6>&- # Revert output to stdout instead of build.log and close file descriptor 6
+                    echo "failed to run $@"
+                    cat build.log && echo "::error title=Failed to stake the blobbers";
+                    echo "exiting blobber stake process..."
+                    exit 1
+                fi
+            }
+            done
+        }
+        retry ./zwallet register --configDir . --config ./zbox_config.yaml --wallet ../ignore > build.log 2>&1
+        for i in {1..6}
+        do
+            retry ./zwallet faucet --methodName "pour" --input "{}" --tokens 2 --configDir . --config ./zbox_config.yaml --wallet ../ignore > build.log 2>&1
+        done
 
-          sleep 10
-          echo "Still waiting for 0Chain deployment to finish..." && ((counter=counter+1));
+        retry ./zwallet getblobbers --silent --configDir . --config ./zbox_config.yaml --wallet ../ignore | grep -o "[a-z0-9]\{64\}" > build.log
+        readarray -t blobber_ids < build.log
+        for id in "${blobber_ids[@]}"
```

Comment:
I would suggest we should check blobber number first. all blobbers should be registered before stake thme.eg
```
while true
do 
   num=$(./zbox ls-blobbers --silent | grep id | wc -l)
   [ "6" -eq $num ]  && break || sleep 10
done
```

Reply:
Added a check to wait for 6 blobbers with a timeout of 60 seconds 👍